Allow lollipops to be worn & bonus longpop

DrDuckedGoose opened the pull request:
## About The Pull Request

Allows lollipops to be inserted into the mask slot. While inserted, the lollipop will be occasionally chewed.

Longpops are a bonus admin item. This item resembles a regular lollipop, with a long straw. Tripping with this will cause severe brute & oxygen damage.

## Why It's Good For The Game

Spessmen have been aggressively risking their health smoking cigarettes to look cool. This no longer has to be the case.

## Testing Photographs and Procedure
Testing process & images included.
<details>
<summary>Screenshots&Videos</summary>

To test the process, insert the lollipop into your mask slot, and wait. After a longer period of time the lollipop will be fully eaten.

![image](https://user-images.githubusercontent.com/40559528/173165115-a2746f6b-6bb0-4421-98e2-0a06336eee5d.png)

![image](https://user-images.githubusercontent.com/40559528/173165119-9c0e5f6a-a507-4693-bb9d-62a016afcf96.png)

![image](https://user-images.githubusercontent.com/40559528/173165130-3d3f91c1-8730-4c8f-acdd-a16ae432ec63.png)

![image](https://user-images.githubusercontent.com/40559528/173551537-627b6042-52d3-4364-b8f8-a27c9b9f63be.png)

![image](https://user-images.githubusercontent.com/40559528/173551572-2b4b2c85-fb13-4e1e-a265-d80992c774e3.png)

</details> 

## Changelog
:cl:
tweak: Lollipops can be inserted into the mask slot
tweak: Lollipops processes when equipped
add: Adds longpop
/:cl:

![image](https://user-images.githubusercontent.com/40559528/173165289-53bd0785-0487-42f5-9547-70b4eaee46b9.png)

DrDuckedGoose committed:
initial - 11 6 22

- Add initial changes
    - Add mask flag to slot flag
    - Add process loop, chews lollipop every few ticks

DrDuckedGoose committed:
Update snacks_other.dm

llol111 commented:
Amasing.

Pirill commented:
First off, great PR.

Second off, I think I'd prefer to just have the lollipops as a cosmetic if you keep them in the mouth slot, it just makes them a bit more versatile rather than just a quirky temporary gimmick accessory.

DrDuckedGoose commented:
> First off, great PR.
> 
> Second off, I think I'd prefer to just have the lollipops as a cosmetic if you keep them in the mouth slot, it just makes them a bit more versatile rather than just a quirky temporary gimmick accessory.

I could extend the use time but, I do believe it should resemble cigarettes, in the sense it has limited use.
Maybe 3x the durability of  cigarettes?

PowerfulBacon commented:
Should choke if you slip

Pirill commented:
> Maybe 3x the durability of cigarettes?

Hmm, maybe! It's just a shame lollipops can only be obtained from mediborgs, but that's probably a thing for another PR. Good job. :)

DrDuckedGoose commented:
> Should choke if you slip

Consider some more testmerges and I'll do it. :trollface:

Archanial commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -438,6 +442,26 @@
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/reagent_containers/food/snacks/lollipop/process(delta_time)
+	tick++ //Trying to use delta_time as a reference didn't work too well here.
```
Use delta time and cool down definitions

DrDuckedGoose replied:
Is the new implementation, ``addtimer()``, alright? It might avoid overhead on ``process()``.

Archanial reviewed (dismissed):
They also need sprites

DrDuckedGoose committed:
Replaces process() with addtimer() - 14 6 22

- Implements addtimer callback
- Increases time between bites.

PowerfulBacon commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -438,6 +443,18 @@
 
+/obj/item/reagent_containers/food/snacks/lollipop/equipped(mob/user, slot)
+	. = ..()
+	chewing = (chewing-1)*-1
```
What the
```suggestion
	chewing = !chewing
```

DrDuckedGoose replied:
:trollface:

PowerfulBacon replied:
Wait this doesnt make any sense, for a start it doesnt check the equipped slot so it will be chewed in hand and why would you toggle chewing on/off when its equipped to mouth.
It should set chewing to true when equipped to the mask slot, then set chewing to false if the chew time fails (allowing the timer to be restarted).

DrDuckedGoose replied:
I think it did this at some point, I might've just forgot to add it back. 
Whoops.

DrDuckedGoose committed:
readability - 14 6 22

DrDuckedGoose committed:
Timers respect stack & equip slot. Also, longpop - 14 6 22

- timers are registered to an id that is deleted and reset respective of equip
- equip respects mask slot
- adds longpop
    - admin item
    - tripping with longpop deals brute & oxy

DrDuckedGoose committed:
Update snacks_other.dm

DrDuckedGoose committed:
Update snacks_other.dm

PowerfulBacon commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -438,6 +445,47 @@
+		deltimer(timer_id)
+	..()
+
+/obj/item/reagent_containers/food/snacks/lollipop/equipped(mob/user, slot)
+	. = ..()
+	if(timer_id)
+		deltimer(timer_id)
+		timer_id = null
+	chewing = (slot == ITEM_SLOT_MASK ? TRUE : FALSE)
+	if(chewing) //Set a timer to chew(), instead of calling chew for the convenience of being able to equip/unequip our pop
+		timer_id = addtimer(CALLBACK(src, .proc/chew), bite_frequency, TIMER_STOPPABLE)	
```
You need a dropped() for this too, for when items are dropped straight to the floor (stripping)

DrDuckedGoose committed:
Edge Case Address - 29 6 22

- Address some edge cases concerning dropped.

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into pop-stick

 Git conflicts, merges master sprites with additions

PowerfulBacon reviewed (approved):
Seems okay
